spec/lib/carto/named_maps/template_spec.rb

```diff
@@ -70,6 +70,37 @@ module NamedMaps
             @template_hash[:layergroup][:layers].second[:id].should eq @carto_layer.id
           end
 
+          describe 'without tooltip or infowindow' do
+            before(:all) do
+              @carto_layer.tooltip = nil
+              @carto_layer.infowindow = nil
+              @carto_layer.save
+
+              @visualization.reload
+
+              template_hash = Carto::NamedMaps::Template.new(@visualization).to_hash
+              @layer_options_hash = template_hash[:layergroup][:layers].second[:options]
+            end
+
+            after(:all) do
+              @carto_layer.tooltip = nil
+              @carto_layer.infowindow = nil
+              @carto_layer.save
+
+              @visualization.reload
+              @layer_options_hash = nil
+            end
+
+            it 'should use cartodb_id as the default interactivity' do
+              @layer_options_hash[:interactivity].should_not be_nil
+              @layer_options_hash[:interactivity].should eq 'cartodb_id'
+            end
+
+            it 'should not contain attributes' do
+              @layer_options_hash[:attributes].should be_nil
+            end
```

Comment:
What about the behavior where `layer_options[:interactivity]` already had a value and it was being overwritten?

That's a kickass spec to be here IMHO

Reply:
Ah wait, you removed that logic? That's what you meant in your description of the update?

Reply:
While `layer_options[:interactivity]` does have an existing value before it is computed by the named maps api, I came across a couple reasons not to use `layer_options[:interactivity]`.
1. All the columns which make up `layer_options[:interactivity]` do seem to exist already in the `layer.infowindow` and `layer.tooltip` properties.  I wrote a query to confirm this.  In fact, it showed that in some cases, `layer_options[:interactivity]` is actually missing some fields which are specified by `layer.tooltip`.  Thus, `layer.infowindow` and `layer.tooltip` should be used as the authority for determining interactivity, not `layer_options[:interactivity]`.
2. The issue only occurs when we have a layer with no `tooltip` or `infowindow`.  Passing an interactivity of `cartodb_id` in this case will prevent this error and also ensure no unnecessary interactivity data is sent to the client from windshaft.

Reply:
For reference, the query to compare `layer.options[:interactivity]` to the expected interactivity based on the logic of the named maps API is [here](https://gist.github.com/tylerparsons/19f7b997443a4d0c56fb071f3a865c5e).

Reply:
Good research!

2. is the sufficient reason for me here: you've found an issue, created tests for it, and implemented the fix. If you have removed the fix and confirmed that the test won't pass without the fix first, it's looking good to me!

Reply:
Yep, I ran the test against the ref before any of these changes to confirm it identified the issue